Declare Python 3.14, 3.15 and Django 6.0, 6.1 support

Stormheg opened the pull request:
~~Not yet declaring support for Django 6.0 official, as the final release is only due in December and I don't want people to be confused by 6.0 appearing in the README under supported versions.~~ _Django 6.0 is now released, updated to add the trove classifier too_

Also dropping all unsupported Django versions to clean up the test matrix - running those tests wastes CI minutes and we are not going to fix issues that crop up from now on with those versions anyway.

~~Since Python 3.14 is not released yet and will fail the CI, I'm marking this as draft. According to [release schedule](https://peps.python.org/pep-0745/) the final release is expected next week, so we should be good to merge this next week.~~ _now released and available_

Stormheg committed:
Drop testing against Django 5.0 and 5.1

Both versions no longer receive official support.

Stormheg committed:
Add testing against Django 6.0

Stormheg committed:
Add Python 3.14 support

Stormheg committed:
Update action versions

Stormheg committed:
Drop Python 3.9 support

alanverresen commented on tox.ini:
```diff
@@ -1,30 +1,29 @@
-    py{310,311,312}-django50,
-    py{310,311,312,313}-django{51,52},
+    py{310,311,312}-django42,
+    py{310,311,312,313}-django52,
```
As of [Django 5.2.8](https://docs.djangoproject.com/en/6.0/releases/5.2/), it also supports Python 3.14.

Stormheg replied:
Added!

alanverresen commented on tox.ini:
```diff
@@ -1,30 +1,29 @@
-    django50: Django~=5.0.0
-    django51: Django~=5.1.0
-    django52: Django~=5.2.0b1
+    django52: Django~=5.2.0
```
If we want to add Django 5.2's support for Python 3.14 to the support matrix, then `5.2.0` should be changed to `5.2.8` to the test matrix.

alanverresen reviewed (dismissed):
I think this is okay as it currently is. If you feel like expanding the test matrix is necessary, I'll try to re-approve the changes as quickly as possible.

mgrdcm commented:
Any reason not to get this checked in @Stormheg?

Stormheg committed:
Add py314-django60 to tox matrix

Stormheg committed:
Update github actions

Stormheg commented:
No reason, just needed a little nudge ;) – thanks @mgrdcm

Stormheg committed:
Add testing against Django 6.1 (release candidate)

mgax commented on tox.ini:
```diff
@@ -1,30 +1,30 @@
-    django52: Django~=5.2.0b1
+    django52: Django~=5.2.0
+    django60: Django~=6.0.0
+    django61: Django~=6.1.0a1
```
By now we can drop Django 5.2 and reference [6.1.b1](https://pypi.org/project/Django/6.1b1/).


```suggestion
    django60: Django~=6.0.0
    django61: Django~=6.1.0b1
```

mgax reviewed (approved):
LGTM 👍 I've left a note about Django versions; we can fix them in this PR or later, I'm fine either way.

Stormheg committed:
Declare Django 6.1 support

Stormheg committed:
Declare Python 3.15 support

Stormheg committed:
Drop Django 4.2 support

Stormheg committed:
Allow pre-release Python 3.15 in CI

Stormheg commented:
@mgax I added Django 6.1 and Python 3.15 support. Since Django 6.1 general release is planned for August and Python 3.15 in October, I propose we target a release date for the v4.2.0 milestone around that time. I've updated the GitHub milestone to reflect.

We should probably merge this PR around that time as to not confuse visitors who are reading the README.

(or merge everything except the README declarations...)

Thoughts?

Andrew-Chen-Wang reviewed (approved):
fwiw, I'm not too worried about this. There doesn't seem to be any django level issues.

Stormheg commented:
Thanks for the review @Andrew-Chen-Wang – lets merge this then 👍
